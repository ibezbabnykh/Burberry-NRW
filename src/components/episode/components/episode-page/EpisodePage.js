import React, { useEffect } from 'react';
import { connect, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';

import { Spinner, ErrorMessage } from 'components/common';
import { loadEpisodeDetails } from '../../store/actions';
import EpisodeDetails from '../episode-details';

const EpisodePage = ({ data, error, loading }) => {
  const { id } = useParams();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loadEpisodeDetails(id));
  }, [id, dispatch]);

  if (loading || !data) {
    return <Spinner />;
  }

  if (error) {
    return <ErrorMessage text={error} />;
  }

  return (
    <div className="episode-page">
      <EpisodeDetails data={data} />
    </div>
  );
};

const mapStateToProps = ({ episode: { data, error, loading } }) => ({
  data, error, loading
})

export default connect(mapStateToProps)(EpisodePage);