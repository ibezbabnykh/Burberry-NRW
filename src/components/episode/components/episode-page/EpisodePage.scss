@import '../../../../assets/style/variables.scss';

.episode-page {
  
}