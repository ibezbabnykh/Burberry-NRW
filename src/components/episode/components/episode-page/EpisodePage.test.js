import { Provider } from 'react-redux';
import EpisodePage from './EpisodePage';

const mockStore = configureStore();

jest.mock('react-router-dom', () => ({
  useParams: () => ({ id: 2 }),
}));

const mockEpisode = {
    id: 1,
    image: {
      medium: 'https://via.placeholder.com/250x140',
      original: 'https://via.placeholder.com/1280x720'
    },
    name: 'episode name',
    summary: 'episode summary'
};

describe('EpisodePage component', () => {
  let store;
  let component;

  const initialState = {
    episode: {
      data: mockEpisode,
      error: null,
      loading: false
    }
  };

  beforeEach(() => {
    store = mockStore(initialState);

    store.dispatch = jest.fn();
 
    component = renderer.create(
      <Provider store={store}>
        <EpisodePage />
      </Provider>
    );
  });

  it('should render', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});