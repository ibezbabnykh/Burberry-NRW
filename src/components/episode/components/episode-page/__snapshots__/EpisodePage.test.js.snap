// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodePage component should render 1`] = `
<div
  className="episode-page"
>
  <article
    className="episode-details"
  >
    <h1
      className="episode-details_title"
    >
      episode name
    </h1>
    <div
      className="episode-details_content"
    >
      <div
        className="episode-details_img"
      >
        <a
          href="https://via.placeholder.com/1280x720"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="episode name"
            src="https://via.placeholder.com/250x140"
            title="episode name"
          />
        </a>
      </div>
      <div
        className="episode-details_description"
      >
        episode summary
      </div>
    </div>
  </article>
</div>
`;
